doc/FAQ.md

```diff
@@ -0,0 +1,104 @@
+# Documentation
+
+## Frequently Asked Questions
+
+### Why is my job running on a single executor? Why am I not seeing any parallelism?
+
+The first thing to check when you see that a Spark job is not being parallelized is to
+determine how many tasks have been generated. To check this look at the UI for your spark
+job and see how many tasks are being run. In the current Shell a small progress bar is shown
+when running stages, the numbers represent (Completed Tasks + Running Tasks) / Total Tasks 
+
+    [Stage 2:=============================================>  (121 + 1) / 200]0
+
+If you see that only a single task has been created this means that the Cassandra Token range has
+not been split into a enough tasks to be well parallelized on your cluster. The number of 
+Spark partitions(tasks) created is directly controlled by the setting `spark.cassandra.input.split.size`.
+This number reflects the approximate number of live Cassandra Partitions in a given Spark partition.
+To increase the number of Spark Partitions decrease this number from the default (100k) to one that
+will sufficiently break up your C* token range. This can also be adjusted on a per cassandraTable basis
+with the function `withReadConf` and specifying a new `ReadConf` object.
+
+If there are more than one tasks but only a single machine is working make sure that the job itself
+has been allocated multiple executor slots to work with. This is set at the time of SparkContext
+creation with `spark.cores.max` in the `SparkConf` and cannot be changed during the job. 
```

Comment:
There can be another reason for lack of parallelism - using `where` with a partition-key predicate. Currently this produces always a single Spark task. We plan to improve this, so that if `where` contained an IN clause with multiple values, there would be a task per partition key value.